src/indexer/api.rs

```diff
@@ -1,258 +1,329 @@
-    match state
-        .write()
+) -> Result<Json<Vec<BlobDb>>, StatusCode> {
+    let tx_hash_bytes = tx_hash.as_bytes();
+    // TODO: Order transaction ?
+    let blobs = sqlx::query_as::<_, BlobDb>("SELECT * FROM blobs WHERE tx_hash = $1")
+        .bind(tx_hash_bytes)
+        .fetch_all(&state)
         .await
-        .blobs
-        .get_with_hash(&tx_hash, blob_index)
-    {
-        Ok(Some(blob)) => Ok(Json(blob)),
-        Ok(None) => Err(StatusCode::NOT_FOUND),
-        Err(_) => Err(StatusCode::INTERNAL_SERVER_ERROR),
+        .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
+
+    match blobs.len() {
+        0 => Err(StatusCode::NOT_FOUND),
+        _ => Ok(Json(blobs)),
     }
 }
 
-pub async fn get_transactions(
-    Query(filters): Query<Filters>,
+pub async fn get_blob(
+    Path((tx_hash, blob_index)): Path<(String, i32)>,
     State(state): State<IndexerState>,
-) -> Result<Json<Vec<Transaction>>, StatusCode> {
-    let (last_height, txs_len) = {
-        let blocks = &state.read().await.blocks;
-        let b = blocks.last();
-        (b.height, b.txs.len())
-    };
-    Ok(Json({
-        filter_iter(
-            state.write().await.transactions.range(
-                TransactionsKey(BlockHeight(0), 0),
-                TransactionsKey(last_height, txs_len),
-            ),
-            filters,
-        )
-    }))
-}
+) -> Result<Json<BlobDb>, StatusCode> {
+    let tx_hash_bytes = tx_hash.as_bytes();
+    let blob =
+        sqlx::query_as::<_, BlobDb>("SELECT * FROM blobs WHERE tx_hash = $1 AND blob_index = $2")
+            .bind(tx_hash_bytes)
+            .bind(blob_index)
+            .fetch_optional(&state)
+            .await
+            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
 
-pub async fn get_last_transaction(
-    State(state): State<IndexerState>,
-) -> Result<Json<Transaction>, StatusCode> {
-    match state.read().await.transactions.last() {
-        Ok(Some(transaction)) => Ok(Json(transaction)),
-        Ok(None) => Err(StatusCode::NOT_FOUND),
-        Err(_) => Err(StatusCode::INTERNAL_SERVER_ERROR),
+    match blob {
+        Some(blob) => Ok(Json(blob)),
+        None => Err(StatusCode::NOT_FOUND),
     }
 }
 
-pub async fn get_transaction(
-    Path((block_height, tx_index)): Path<(BlockHeight, usize)>,
-    State(state): State<IndexerState>,
-) -> Result<Json<Transaction>, StatusCode> {
-    match state.write().await.transactions.get(block_height, tx_index) {
-        Ok(Some(transactions)) => Ok(Json(transactions)),
-        Ok(None) => Err(StatusCode::NOT_FOUND),
-        Err(_) => Err(StatusCode::INTERNAL_SERVER_ERROR),
-    }
-}
+// Proofs
+// pub async fn get_last_proof(State(state): State<IndexerState>) -> Result<Json<Proof>, StatusCode> {
+//     let proof = sqlx::query_as::<_, Proof>("SELECT * FROM proofs ORDER BY id DESC LIMIT 1")
+//         .fetch_optional(&state)
+//         .await
+//         .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
```

Comment:
commented code ?

Reply:
Not sure if we'll actually use them in the end... Wanted to keep it commented while it's not decided... Maybe we could simply remove it